bom/pom.xml

```diff
@@ -21,55 +21,70 @@
-        <latencyUtilsVersion>2.0.3</latencyUtilsVersion>
-        <!-- Neo4j supports different versions of Cypher -->
-        <neo4jCypherCompiler23Version>2.3.4</neo4jCypherCompiler23Version>
-        <neo4jCypherCompiler30Version>${neo4jVersion}</neo4jCypherCompiler30Version>
+        <!--
+            These dependencies should be aligned with the ones from the Wildfly version we support
+            See the version.wildfly property in the hibernate-ogm-parent pom
+            See http://search.maven.org/#search|gav|1|g%3A"org.wildfly"%20AND%20a%3A"wildfly-parent"
+        -->
         <hibernateVersion>5.0.9.Final</hibernateVersion>
         <hibernateSearchVersion>5.5.4.Final</hibernateSearchVersion>
-        <hibernateParserVersion>1.3.0.Final</hibernateParserVersion>
         <hibernateCommonsAnnotationsVersion>5.0.1.Final</hibernateCommonsAnnotationsVersion>
-        <!-- Careful: Lucene is shared across Neo4J dependencies, Infinispan and Hibernate Search -->
-        <luceneVersion>5.5.2</luceneVersion>
-        <narayanaVersion>5.2.14.Final</narayanaVersion>
-        <jbossJtaAPIVersion>1.0.0.Final</jbossJtaAPIVersion>
         <jbossLoggingVersion>3.2.1.Final</jbossLoggingVersion>
         <jbossLoggingProcessorVersion>1.2.0.Final</jbossLoggingProcessorVersion>
-        <slf4jVersion>1.7.5</slf4jVersion>
-        <resteasyVersion>3.0.14.Final</resteasyVersion>
-        <jacksonVersion>2.5.4</jacksonVersion>
-        <parboiledVersion>1.1.7</parboiledVersion>
-        <scalaLibraryVersion>2.11.8</scalaLibraryVersion>
-        <concurrentlinkedhashmapVersion>1.4.2</concurrentlinkedhashmapVersion>
-        <commonsLang3Version>3.3.2</commonsLang3Version>
-        <asmVersion>5.0.3</asmVersion>
+        <infinispanVersion>8.2.3.Final</infinispanVersion>
+        <classmateVersion>1.3.1</classmateVersion>
+        <resteasyVersion>3.0.18.Final</resteasyVersion> <!-- Used by CouchDB and Neo4j remote -->
+        <jacksonVersion>2.7.4</jacksonVersion>
+        <slf4jVersion>1.7.7</slf4jVersion> <!-- Note: Wildfly 10.1.0.CR1 uses a custom jboss version -->
+        <netty4Version>4.0.33.Final</netty4Version>
+        <jbossJtaAPIVersion>1.0.0.Final</jbossJtaAPIVersion>
+        <narayanaVersion>5.3.3.Final</narayanaVersion>
+        <!-- Careful: Lucene is shared across Neo4J dependencies, Infinispan and Hibernate Search -->
+        <luceneVersion>5.5.2</luceneVersion>
+
+        <!-- Various dependencies -->
+        <hibernateHqlParserVersion>1.3.0.Final</hibernateHqlParserVersion>
+        <parboiledVersion>1.1.7</parboiledVersion> <!-- needs to be aligned with Neo4j version but is also used for MongoDB -->
+        <asmVersion>5.0.3</asmVersion> <!-- used for Parboiled and Neo4j -->
+
+        <!-- Ehcache -->
+        <ehcacheVersion>2.6.11</ehcacheVersion>
+
+        <!-- MongoDB -->
+        <mongodbDriverVersion>3.3.0</mongodbDriverVersion>
+        <mongodbVersion>3.2.8</mongodbVersion>
```

Comment:
It still works with 3.0. The only test failing is the one testing the index as they changed the textIndexVersion from 2 to 3. Other than that, there are no incompatible changes.